libcxx/include/__utility/small_buffer.h

```diff
@@ -75,7 +75,7 @@ class __small_buffer {
   template <class _Stored>
   _LIBCPP_HIDE_FROM_ABI void __dealloc() noexcept {
     if constexpr (!__fits_in_buffer<_Stored>)
-      ::operator delete[](*reinterpret_cast<void**>(__buffer_), sizeof(_Stored), align_val_t{alignof(_Stored)});
+      std::__libcpp_deallocate((void*)__buffer_, _BufferSize * sizeof(byte), alignof(_Stored));
```

Comment:
This is incorrect, since you're deallocating the buffer itself (which is a member variable), not the pointer contained in the buffer. The previous `reinterpret_cast` must be kept.

Reply:
Thanks for caching it.